### ISSUE TITLE
"Connection closed before message completed" error.

### ISSUE THREAD

**Issue by ghuwe** [Morganamilo/paru#167]
I get the following error on running paru to update AUR packages:

`error: error sending request for url (https://aur.archlinux.org/rpc?{a_bunch_of_args}): connection closed before message completed`

This happens randomly. Sometimes it works perfectly. Sometimes it doesn't.

I suspected it happened because of my networking, but it seems to be unrelated. Anecdotally, it works with very a very unstable and slow network, and fails when it's fast and stable.

This doesn't happen with yay, could it be related to a Rust library difference in relation to a Go library?

**Comment by Morganamilo** [comment]
From googling https://github.com/hyperium/hyper/issues/2136

**Comment by ghuwe** [comment]
It is happening again.

**Comment by benvonh** [comment]
I found I get this error when trying to install certain packages. For example, I tried to install `wlr-randr` but it froze on `Resolving dependencies...` for 2 minutes before producing that error. However, when I used yay it found the package immediately and said that PGP keys needed importing before installing it. After doing that, running `paru -S wlr-randr` works, so maybe it's to do with PGP keys?
